lib/nn-syntax/parser.lr

```diff
@@ -269,8 +269,8 @@ table_join : Ast.JoinTable = {
 stream : Ast.Stream = {
     'timer' 'base' '=' base:value ',' 'interval' '=' interval:value => new Ast.Stream.Timer($.location, base, interval, null, null);
     'timer' 'base' '=' base:value ',' 'interval' '=' interval:value ',' 'frequency' '=' frequency:value => new Ast.Stream.Timer($.location, base, interval, frequency, null);
-    'attimer' 'time' '=' time:value => new Ast.Stream.AtTimer($.location, time instanceof Ast.ArrayValue ? time.value : [time], null, null);
-    'attimer' 'time' '=' time:value ',' 'expiration_date' '=' expiration_date:value => new Ast.Stream.AtTimer($.location, time instanceof Ast.ArrayValue ? time.value : [time], expiration_date, null);
+    'attimer' 'time' '=' time:value => new Ast.Stream.AtTimer($.location, time instanceof Ast.ArrayValue ? time.value : [time], null, false, null);
+    'attimer' 'time' '=' time:value ',' 'expiration_date' '=' expiration_date:value ',' 'once' '=' once:value => new Ast.Stream.AtTimer($.location, time instanceof Ast.ArrayValue ? time.value : [time], expiration_date, once, null);
```

Comment:
This change is not correct because it makes the `once` parameter required if `expiration_date` is specified.
TBQH I wouldn't touch the legacy syntax because this is a new feature, but if we do add the syntax, we should do it in a compatible way.

Reply:
Is the parser.lr just legacy? If so, I can just remove this then.

Reply:
lib/nn-syntax/parser.lr is legacy, lib/new-syntax/parser.lr is current - but the latter should need no changes because timers are treated like regular function calls, and which parameters are allowed is solved by typechecking not syntax.